src/common/mfu_io.c

```diff
@@ -1218,12 +1159,39 @@ int daos_open(const char* file, int flags, mode_t mode, mfu_file_t* mfu_file)
     if (parent == NULL) {
         rc = -1;
     } else {
-        rc = dfs_open(mfu_file->dfs, parent, name,
-                      dfs_mode, flags,
-                      0, 0, NULL, &(mfu_file->obj));
-        if (rc) {
-            MFU_LOG(MFU_LOG_ERR, "dfs_open %s failed (%d %s)",
-                    name, rc, strerror(rc));
+        mode_t obj_mode;
+        rc = dfs_lookup_rel(mfu_file->dfs, parent, name, O_RDWR, &(mfu_file->obj),
+                            &obj_mode, NULL);
+        if (!rc) {
+            /* lookup found an obj */
+            if (flags & O_CREAT && flags & O_EXCL) {
+                /* ... but it should not have */
+                dfs_release(mfu_file->obj);
+                errno = EEXIST;
+                rc = -1;
+            } else if (!S_ISREG(obj_mode)) {
+                /* ... but the obj isn't a file */
+                dfs_release(mfu_file->obj);
+                MFU_LOG(MFU_LOG_ERR, "Invalid entry type (not a file)");
+                errno = EINVAL;
+                rc = -1;
+            } else {
+                /* good */
```

Comment:
@daltonbohning is the empty else block necessary?

Reply:
@dsikich No, it's not necessary. I adapted this style from somewhere else in this repository... though now I'm not sure where.
I can remove it if you like, but I think we'll probably convert the cascading if/else statements into the "goto" style when we move them to daos, since that seems to be the standard in daos.

Reply:
@daltonbohning if you prefer it I don't mind